services/scoop/scoop.service.js

```diff
@@ -0,0 +1,95 @@
+'use strict'
+
+const Joi = require('@hapi/joi')
+const { NotFound } = require('..')
+const {
+  ConditionalGithubAuthV3Service,
+} = require('../github/github-auth-service')
+const { fetchJsonFromRepo } = require('../github/github-common-fetch')
+const { renderVersionBadge } = require('../version')
+
+const gitHubRepoRegExp = new RegExp(
+  'https://github.com/(?<user>.*)/(?<repo>.*)/?'
+)
+const bucketsSchema = Joi.object()
+  .pattern(/.+/, Joi.string().pattern(gitHubRepoRegExp).required())
+  .required()
+const scoopSchema = Joi.object({
+  version: Joi.string().required(),
+}).required()
+const queryParamSchema = Joi.object({
+  bucket: Joi.string(),
+})
+
+module.exports = class ScoopVersion extends ConditionalGithubAuthV3Service {
+  static get category() {
+    return 'version'
+  }
+
+  static get route() {
+    return {
+      base: 'scoop/v',
+      pattern: ':app',
+      queryParamSchema,
+    }
+  }
+
+  static get examples() {
+    return [
+      {
+        title: 'Scoop Version',
+        namedParams: { app: 'ngrok' },
+        staticPreview: this.render({ version: '2.3.35' }),
+      },
+      {
+        title: 'Scoop Version (extras bucket)',
+        namedParams: { app: 'dnspy' },
+        queryParams: { bucket: 'extras' },
+        staticPreview: this.render({ version: '6.1.4' }),
+      },
+    ]
+  }
+
+  static get defaultBadgeData() {
+    return { label: 'scoop' }
+  }
+
+  static render({ version }) {
+    return renderVersionBadge({ version })
+  }
+
+  async handle({ app }, queryParams) {
+    const buckets = await fetchJsonFromRepo(this, {
+      schema: bucketsSchema,
+      user: 'lukesampson',
+      repo: 'scoop',
+      branch: 'master',
+      filename: 'buckets.json',
+    })
+    const bucket = queryParams.bucket || 'main'
+    const bucketUrl = buckets[bucket]
+    if (!bucketUrl) {
+      throw new NotFound({ prettyMessage: `bucket "${bucket}" not found` })
+    }
+    const {
+      groups: { user, repo },
+    } = gitHubRepoRegExp.exec(bucketUrl)
+    try {
+      const { version } = await fetchJsonFromRepo(this, {
```

Comment:
I'm only vaguely familiar with scoop and don't know anything about buckets/etc. I take it the various buckets listed in https://github.com/lukesampson/scoop/blob/master/buckets.json are each registries of sorts where apps can be listed for consumption and thus installed via `scoop`.

If that's correct, do you know how dynamic that bucket list is? I'm assuming that the vast majority of apps would be registered/listed under the `main` bucket and I'm wondering if we can avoid having to make 2 separate API calls for at least the majority of cases.

Reply:
Also, is it possible for additional "3rd party" and/or private buckets? (could I have my own private/internal bucket within a corporate network for example)

Reply:
Maybe we could simply do that second API call once and cache the result for all subsequent calls? Given that Shields instances are redeployed on a regular basis and given that the file changes rarely, we probably don't need to implement any fancy refresh strategy.

Reply:
> we probably don't need to implement any fancy refresh strategy.

Not sure what you were referring to by fancy refresh (we may be thinking the same thing), but I'm simply wondering if we can go ahead make the one API call for the app under https://github.com/ScoopInstaller/Main/tree/master/bucket/foo.json (assuming that's the base default/most common collection of apps) in the majority of cases instead of always first calling https://github.com/lukesampson/scoop/blob/master/buckets.json.

Reply:
I was thinking keep most of the implementation as is, but do the request to https://github.com/lukesampson/scoop/blob/master/buckets.json only once and store its return value in a field on first successful retrieval.

Reply:
What are your thoughts on my above suggestion @0xced? :)